engine/api/src/org/apache/cloudstack/engine/subsystem/api/storage/PrimaryDataStoreDriver.java

```diff
@@ -23,18 +23,21 @@
 
 import com.cloud.host.Host;
 import com.cloud.storage.StoragePool;
-import com.cloud.storage.Volume;
 
 public interface PrimaryDataStoreDriver extends DataStoreDriver {
-    public ChapInfo getChapInfo(VolumeInfo volumeInfo);
+    public ChapInfo getChapInfo(DataObject dataObject);
 
     public boolean grantAccess(DataObject dataObject, Host host, DataStore dataStore);
 
     public void revokeAccess(DataObject dataObject, Host host, DataStore dataStore);
 
     // intended for managed storage (cloud.storage_pool.managed = true)
     // if not managed, return volume.getSize()
-    public long getVolumeSizeIncludingHypervisorSnapshotReserve(Volume volume, StoragePool storagePool);
+    public long getDataObjectSizeIncludingHypervisorSnapshotReserve(DataObject dataObject, StoragePool storagePool);
+
+    // intended for zone-wide primary storage that is capable of storing a template once and using it in multiple clusters
+    // if not this kind of storage, return 0
+    public long getBytesRequiredForTemplate(TemplateInfo templateInfo, StoragePool storagePool);
```

Comment:
This is not a proper documentation. 
I would suggest changing from Java comments style to Javadoc.